src/js/utils/string-utils.js

```diff
@@ -0,0 +1,59 @@
+/**
+ * Useful functions for dealing with strings.
+ */
+(function(define) {
+    'use strict';
+    define(['underscore', './string-utils.js'], function(_, StringUtils) {
+        var interpolate;
+
+        /**
+         * Returns a string created by interpolating the provided parameters.
+         *
+         * The HTML text is provided as a tokenized format string where parameters
+         * are indicated via curly braces, e.g. 'Hello {name}'. These tokens are
+         * replaced by the parameter value of the same name.
+         *
+         * Parameter values will be rendered using their toString methods and then
+         * HTML-escaped. The only exception is that instances of the class HTML
```

Comment:
Is this HTML-escaping at all? I thought this method is just about interpolation.